slick/src/main/scala/slick/jdbc/JdbcTypesComponent.scala

```diff
@@ -164,12 +173,218 @@ trait JdbcTypesComponent extends RelationalTypesComponent { self: JdbcProfile =>
       override def valueToSQLLiteral(v: Char) = stringJdbcType.valueToSQLLiteral(String.valueOf(v))
     }
 
+    /**
+      * Use [[ZonedDateTimeJdbcType]] or [[OffsetDateTimeJdbcType]] or [[LocalTimeJdbcType]]
+      * or [[OffsetTime]] or [[LocalDateTime]] or [[LocalDate]] or [[Instant]] instead.
+      */
+    @deprecated(message = "java.util.Date is deprecated, use java.time.LocalDateTime instead.", since = "3.2.0")
     class DateJdbcType extends DriverJdbcType[Date] {
       def sqlType = java.sql.Types.DATE
       def setValue(v: Date, p: PreparedStatement, idx: Int) = p.setDate(idx, v)
       def getValue(r: ResultSet, idx: Int) = r.getDate(idx)
       def updateValue(v: Date, r: ResultSet, idx: Int) = r.updateDate(idx, v)
-      override def valueToSQLLiteral(value: Date) = "{d '"+value.toString+"'}"
+      override def valueToSQLLiteral(value: Date) = s"{d '${value.toString}'}"
+    }
+
+    class LocalDateJdbcType extends DriverJdbcType[LocalDate] {
+      // TODO: Use native Instant JDBC methods as soon they are implemented.
+      // https://java.net/jira/browse/JPA_SPEC-63
+      override def sqlType : Int = java.sql.Types.DATE
+      override def setValue(v: LocalDate, p: PreparedStatement, idx: Int) : Unit = {
+        p.setDate(idx, if (v == null) null else Date.valueOf(v))
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalDate = {
+        r.getDate(idx) match {
+          case null => null
+          case date => date.toLocalDate
+        }
+      }
+      override def updateValue(v: LocalDate, r: ResultSet, idx: Int) : Unit = {
+        r.updateDate(idx, if (v == null) null else Date.valueOf(v))
+      }
+      override def valueToSQLLiteral(value: LocalDate) = {
+        value match {
+          case null => "NULL"
+          case _ =>  s"'${value.toString}'"
+        }
+      }
+    }
+
+    class LocalTimeJdbcType extends DriverJdbcType[LocalTime] {
+      override def sqlType : Int = {
+        /**
+         * [[LocalTime]] will be persisted as a [[java.sql.Types.VARCHAR]] in order to
+         * avoid losing precision, because the ANSI SQL type ([[java.sql.Types.TIME]]) stores
+         * TIME with second precision, while [[LocalTime]] stores it with a millisecond one.
+         */
+        java.sql.Types.VARCHAR
+      }
+      override def setValue(v: LocalTime, p: PreparedStatement, idx: Int) : Unit = {
+        p.setString(idx, if (v == null) null else v.toString)
+      }
+      override def getValue(r: ResultSet, idx: Int) : LocalTime = {
+        r.getString(idx) match {
+          case null => null
+          case iso8601String => LocalTime.parse(iso8601String)
+        }
+      }
+      override def updateValue(v: LocalTime, r: ResultSet, idx: Int) = {
+        r.updateString(idx, if (v == null) null else v.toString)
+      }
```

Comment:
https://github.com/slick/slick/blob/2cb76c3fd147a2d5c6062b563bc8f168fee0a6ce/slick/src/main/scala/slick/jdbc/SpecializedJdbcResultConverters.fm#L23
Since the code above not like other place
https://github.com/slick/slick/blob/3b3bd36c93c6d9c63b0471ff4d8409f913954b2b/slick/src/main/scala/slick/jdbc/JdbcResultConverter.scala#L13
So all jdbcType can not return null even if you override def wasNull, so I think you can ignore the null value.